### ISSUE TITLE
Additional devices in the selection

### ISSUE THREAD

**Issue by Victini378** [Belphemur/SoundSwitch#1045]
### What happened?

Very often it takes devices that have not been selected (as in the attached photo)
![image](https://user-images.githubusercontent.com/44738063/199762349-6962837b-5bd5-4bf8-8177-b36d4f2b0d10.png)

### Step to reproduce

It happens I think during the reboot but not always

### Version

I don't know, but for a few months now. I always keep the software up to date

### What version of Windows are you seeing the problem on?

Windows 10

**Comment by C0ntroller** [comment]
I have the same problem (latest release 6.4.3, Win11) and for me an easy way to reproduce is updating my NVidia drivers. As HDMI audio drivers get reinstalled, I suspect the devices get deleted and readded with a new ID? Seems hard to prevent this.

@Victini378 You could try and deselect "Automatically select new devices" in the settings, maybe this helps?

**Comment by Victini378** [comment]
> I have the same problem (latest release 6.4.3, Win11) and for me an easy way to reproduce is updating my NVidia drivers. As HDMI audio drivers get reinstalled, I suspect the devices get deleted and readded with a new ID? Seems hard to prevent this.
> 
> @Victini378 You could try and deselect "Automatically select new devices" in the settings, maybe this helps?

However, this option does not solve the user experience. It would take a selection of the outputs that I never want. So the new headphones are automatically selected, but the various outputs that we "hate" we can ignore and never see them.

**Comment by C0ntroller** [comment]
> It would take a selection of the outputs that I never want.  So the new headphones are automatically selected, but the various outputs that we "hate" we can ignore and never see them.

You are right, I missunderstood what the switch will do.

Then maybe the devs could add another switch that "locks" the SoundSwitch selection? No new devices are added automatically? (This is what I just thought the switch would do.)

**Comment by Belphemur** [comment]
Hello,

As suggested unchecking the "Automatically add new devices" is the way to go.

Also, SoundSwitch relies on Name of devices or Ids.
As long as either of those match, it's considered a selected device.

What I suggest is you update to the latest version manually, ask it to delete your settings then rename the device that are handled by your graphic card.

Select the devices you want in SoundSwitch. That should do it.

Sent from Nine
________________________________

**Comment by Victini378** [comment]
Yes, but it still doesn't solve the problem.
What I mean is that if I plug in a new audio device I'd like the SoundSwitch to plug it in automatically. So the way to go would be to also have a pick list for devices I want to block from automatically popping into my list.
I hope I made you understand the problem.

**Comment by maltaethirion** [comment]
Hello,
It happens for me as well.
After reboot sometimes it adds my HDMI monitor (which doesn't have speakers at all) to the selection of devices.

**Comment by Victini378** [comment]
> 

In that case the monitor could have an audio output (probably with a jack type socket)

**Comment by rahmnstein** [comment]
Yes, everytime I reboot, all playback devices are selected. Windows 11.
